akka-projection-alpakka-kafka/src/test/scala/akka/projection/scaladsl/kafka/AtLeastOnceRunnerSpec.scala

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (C) 2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.projection.scaladsl.kafka
+
+import akka.Done
+import akka.kafka.Subscriptions
+import akka.kafka.scaladsl.Consumer
+import akka.projection.testkit.{DBIO, DbProjectionHandler, InMemoryRepository}
+import akka.projection.scaladsl.Projection
+import akka.stream.scaladsl.{Keep, Sink}
+import akka.stream.testkit.scaladsl.StreamTestKit.assertAllStagesStopped
+import akka.stream.testkit.scaladsl.TestSink
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+class AtLeastOnceRunnerSpec extends TestcontainersKafkaSpec {
+
+  implicit val patience: PatienceConfig = PatienceConfig(30.seconds, 500.millis)
+
+
+  "Kafka Projection using at-least-once delivery" must {
+
+    "deliver events to ProjectionHandler" in assertAllStagesStopped {
+
+      val Messages = "abc" :: "def" :: "ghi" :: Nil
+      val topic1 = createTopic(1)
+      val group1 = createGroupId(1)
+      Await.result(produceString(topic1, Messages), remainingOrDefault)
+
+      val repository = new InMemoryRepository[String]
+      val handler = new DbProjectionHandler[String] {
+        override def handleEvent(event: String): DBIO[Done] = {
+          repository.save(event)
+        }
+      }
+
+      // this is how the API looks like when using it directly. 
+      //      Projection(
+      //        KafkaSourceProviders.committableSource(consumerDefaults.withGroupId(group1), Subscriptions.topics(topic1)),
+      //        KafkaExtractors.committableSource[String, String],
+      //        KafkaProjectionRunners.atLeastOnceRunner,
+      //        handler.asAsyncHandler
+      //      ).start()
+
+      val projection =
+        KafkaProjections
+          .committableSource(consumerDefaults.withGroupId(group1), Subscriptions.topics(topic1))
+          .withAtLeastOnce
+          .withEventHandler(handler.asAsyncHandler)
```

Comment:
The commented block shows how the API looks like when using it directly. There is a bunch of unnecessary concepts that power-users may want (when defined their own SourceProviders, for instance). 

Each projection implementation can define its own builder to cover the most common use cases. For instance, `KafkaProjections` is defining a fluent build in which you can configure your Consumer source and the event handler. 

Basically, here is from where I read, here is where I write.

The `withAtLeastOnce` is optional. I only added here to demo the API. We could also write:

```scala
KafkaProjections
          .committableSource(consumerDefaults.withGroupId(group1), Subscriptions.topics(topic1))
          .withEventHandler(handler.asAsyncHandler)
```